clang/include/clang/Sema/SemaARM.h

```diff
@@ -37,15 +37,20 @@ class SemaARM : public SemaBase {
                             /// flags. Do Sema checks for the runtime mode.
   };
 
+  bool CheckImmediateArg(CallExpr *TheCall, unsigned CheckTy, unsigned ArgIdx,
+                         unsigned EltBitWidth, unsigned VecBitWidth);
   bool CheckARMBuiltinExclusiveCall(unsigned BuiltinID, CallExpr *TheCall,
                                     unsigned MaxWidth);
   bool CheckNeonBuiltinFunctionCall(const TargetInfo &TI, unsigned BuiltinID,
                                     CallExpr *TheCall);
+  bool
+  ParseNeonImmChecks(CallExpr *TheCall,
+                     SmallVector<std::tuple<int, int, int, int>, 2> &ImmChecks,
```

Comment:
Perhaps `std::tuple<int, int, int ...` warrants its own type name, no strong opinion either way.

Reply:
I would have done this if the type name could be shared SVE/SME and NEON, but this isn't the case as we omit the last value from the tuple in SME/SVE as it is always 128.